pkg/oci/utils.go

```diff
@@ -286,6 +286,31 @@ func (spec *CompatOCISpec) PodID() (string, error) {
 	return "", fmt.Errorf("Could not find pod ID")
 }
 
+// TODO: calculate number of VCPUs from OCI spec
+func vmConfig(ocispec CompatOCISpec, config RuntimeConfig) (vc.Resources, error) {
+	if ocispec.Linux == nil ||
+		ocispec.Linux.Resources == nil ||
+		ocispec.Linux.Resources.Memory == nil ||
+		ocispec.Linux.Resources.Memory.Limit == nil {
+		return config.VMConfig, nil
+	}
+
+	memBytes := *ocispec.Linux.Resources.Memory.Limit
+
+	// round up memory to 1MB
+	mem := uint((memBytes + (1024*1024 - 1)) / (1024 * 1024))
+	if mem <= 0 {
```

Comment:
We should check `memBytes` instead of `mem`, that's the one which could be negative. You should make this check right after you retrieved `memBytes` value.